src/gc.h

```diff
@@ -11,20 +11,24 @@
 
 #define V7_DISABLE_GC
 
-#define MARK(p) (* (uintptr_t *) &(p) |= 1)
+#define MARK(p) (* (unsigned long *) &(p) |= 1)
```

Comment:
The GC works by tagging the first pointers of a structure, not val_t.

```c
struct v7_object {
  /* First HIDDEN property in a chain is an internal object value */
  struct v7_property *properties;
  struct v7_object *prototype;
};
```

With this change when you mark an object, it will set the LSB of the object's prototype, which means that it will be traversed and it will crash.